src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js

```diff
@@ -91,13 +94,83 @@
                 return item.name === dimension;
             })
         }
+        function getStartDateRange() {
+            var date = new Date();
+            var config = vm.currentDynamicDateRangeConfig;
+            if (config.toDate) {
+                date = moment(date).startOf(config.toDate).toDate();
+                return date;
+            }
+            return null;
+        }
+        function getStartDateRangeInterval(filterTime) {
+
+            var activeFilter = vm.dynamicDateRangeConfig.filter(function (item) {
+                return item.title === vm.currentDynamicDateRangeConfig;
+            });
+
+            var config = vm.currentDynamicDateRangeConfig;
+            if (config.toDate) {
+                return null;
+            } else if (activeFilter[0].isCustom) {
+                return filterTime + ' ' + activeFilter[0].unit;
+            } else if (activeFilter[0].period.days) {
+                return activeFilter[0].period.days + ' days';
+            } else if (activeFilter[0].period.months) {
+                return activeFilter[0].period.months + ' months';
+            }
+            return null;
+        }
+        function formatDate(date) {
+            if (!date) {
+                return null;
+            }
+            return filterParametersService.dateToString(date);
+        }
+
+        function resetTimezone(date) {
+            if (!date) {
```

Comment:
These all are duplicate functions already exist in date range component.. Try to create date util service and place them in separate file